kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/testplugins/ClientAuthAwareLawyerFilter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.testplugins;
+
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.security.auth.x500.X500Principal;
+
+import org.apache.kafka.common.header.internals.RecordHeader;
+
+import io.kroxylicious.proxy.filter.FilterContext;
+import io.kroxylicious.proxy.tls.ClientTlsContext;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A filter that adds {@linkplain FilterContext#clientTlsContext() client-facing TLS context}-dependent headers to produced records.
+ * Tests can consume the produced records and assert that those records have the expected header values.
+ */
+public class ClientAuthAwareLawyerFilter
+        extends AbstractProduceHeaderInjectionFilter {
+
+    @NonNull
+    private static String headerName(String hashtag) {
+        return ClientAuthAwareLawyerFilter.class.getSimpleName() + hashtag;
+    }
+
+    public static final String HEADER_KEY_CLIENT_TLS = headerName("#clientTlsContext.isPresent");
+    public static final String HEADER_KEY_CLIENT_TLS_PROXY_X500PRINCIPAL_NAME = headerName("#clientTlsContext.proxyServerCertificate.principalName");
+    public static final String HEADER_KEY_CLIENT_TLS_CLIENT_X500PRINCIPAL_NAME = headerName("#clientTlsContext.clientCertificate.principalName");
+
+    private static final Map<String, Function<FilterContext, byte[]>> HEADERS = Map.of(
+            HEADER_KEY_CLIENT_TLS,
+            context -> context.clientTlsContext().isPresent() ? new byte[]{ 1 } : new byte[]{ 0 },
```

Comment:
nit: I'd rather see a boolean in the testing code, maybe serialize as `Boolean.toString(blah).getBytes(charset)` and have a matching assertion that can take a boolean?

Reply:
kind of a big assumption for the generic Header assert library to make isn't it, hmm, okay I see that might not be such a good idea